presto-main/src/test/java/com/facebook/presto/sql/query/TestFilteredAggregations.java

```diff
@@ -124,6 +124,17 @@ public void testGroupAll()
                 "VALUES (BIGINT '3', BIGINT '6')");
     }
 
+    @Test
+    public void testSetAggWithNulls()
+    {
+        assertions.assertQuery(
+                "SELECT y, set_agg(y) FILTER (WHERE x = 1) FROM (SELECT 1 x, 2 y UNION ALL SELECT NULL x, 20 y UNION ALL SELECT 1 x, NULL y) GROUP BY y ORDER BY y",
+                "VALUES (INTEGER '2', ARRAY[INTEGER '2']), (INTEGER '20', CAST(NULL AS ARRAY<INTEGER>)), (CAST(NULL AS INTEGER), ARRAY[CAST(NULL AS INTEGER)])");
+        assertions.assertQuery(
+                "SELECT y, set_agg(IF(x = 1,y)) FROM (SELECT 1 x, 2 y UNION ALL SELECT NULL x, 20 y UNION ALL SELECT 1 x, NULL y) GROUP BY y ORDER BY y",
+                "VALUES (INTEGER '2', ARRAY[INTEGER '2']), (INTEGER '20', ARRAY[CAST(NULL AS INTEGER)]), (CAST(NULL AS INTEGER), ARRAY[CAST(NULL AS INTEGER)])");
```

Comment:
Add a couple of more tests for weird aggs like approx_set, set_union, map_union and map_union_sum.

Reply:
Done.